internal/contentmanager/contentmanager.go

```diff
@@ -0,0 +1,203 @@
+			ul := &unstructured.UnstructuredList{}
+			ul.SetGroupVersionKind(gvk.GroupVersion().WithKind(listKind))
+
+			scheme.AddKnownTypeWithName(gvk, u)
+			scheme.AddKnownTypeWithName(gvk.GroupVersion().WithKind(listKind), ul)
+			// Adding the common meta schemas to the scheme for the GroupVersion
+			// is necessary to ensure the scheme is aware of the different operations
+			// that can be performed against the resources in this GroupVersion
+			metav1.AddToGroupVersion(scheme, gvk.GroupVersion())
+		}
+	}
+
+	return scheme, nil
+}
+
+// ManageContent configures a controller-runtime cache.Cache and establishes watches for the provided resources.
+// It utilizes the provided ClusterExtension to set a DefaultLabelSelector on the cache.Cache
+// to ensure it is only caching and reacting to content that belongs to the ClusterExtension.
+// For each client.Object provided, a new source.Kind is created and used in a call to the Watch() method
+// of the provided controller.Controller to establish new watches for the managed resources.
+func (i *instance) ManageContent(ctx context.Context, ctrl controller.Controller, ce *v1alpha1.ClusterExtension, objs []client.Object) error {
+	// Return nil if the objs slice is empty before any further action
+	if len(objs) == 0 || ce == nil || ctrl == nil {
+		return nil
+	}
+
+	cfg, err := i.rcm(ctx, ce, i.baseCfg)
+	if err != nil {
+		return fmt.Errorf("getting rest.Config for ClusterExtension %q: %w", ce.Name, err)
+	}
+
+	scheme, err := buildScheme(objs)
+	if err != nil {
+		return fmt.Errorf(
+			"building scheme for %s; %w",
+			ce.GetName(),
+			err,
+		)
+	}
+
+	tgtLabels := labels.Set{
+		oclabels.OwnerKindKey: v1alpha1.ClusterExtensionKind,
+		oclabels.OwnerNameKey: ce.GetName(),
+	}
+
+	c, err := cache.New(cfg, cache.Options{
+		Scheme:               scheme,
+		DefaultLabelSelector: tgtLabels.AsSelector(),
+	})
+	if err != nil {
+		return fmt.Errorf("creating cache for ClusterExtension %q: %w", ce.Name, err)
+	}
+
+	for _, obj := range objs {
+		err = ctrl.Watch(
+			source.Kind(
+				c,
+				obj,
+				handler.TypedEnqueueRequestForOwner[client.Object](
+					scheme,
+					i.mapper,
+					ce,
+				),
+			),
+		)
+		if err != nil {
+			return fmt.Errorf("creating watch for ClusterExtension %q managed resource %s: %w", ce.Name, obj.GetObjectKind().GroupVersionKind(), err)
+		}
+	}
+
+	// TODO: Instead of stopping the existing cache and replacing it every time
+	// we should stop the informers that are no longer required
+	// and create any new ones as necessary. To keep the initial pass
+	// simple, we are going to keep this as is and optimize in a follow-up.
+	// Doing this in a follow-up gives us the opportunity to verify that this functions
+	// as expected when wired up in the ClusterExtension reconciler before going too deep
+	// in optimizations.
+	if data, ok := i.extensionCaches[ce.GetName()]; ok {
+		data.Cancel()
+	}
+
+	ctx, cancel := context.WithCancel(ctx)
+	go c.Start(ctx) //nolint:errcheck
```

Comment:
No error check? What happens when `Start()` returns an error and we don't check it?

Reply:
I was scratching my head on how to resolve this problem and couldn't come up with anything because the `Start()` method is blocking. The only thing I could think of is _maybe_ something like:
```go
errChan := make(chan error)
go func(){
  err := c.Start(ctx)
  if err != nil {
    errChan <- err
  }
}()
// handle errors down here
```
But I'm not sure what would happen if the function that created the channel had already exited.

I decided to go the route of not doing an error check here, but waiting for the cache to sync as the next step. If the cache could not sync we cancel the context and return an error. Otherwise the cache has started and we should be good to go.

I'm open to suggestions for better ways to handle this, so if you have some ideas here I'm all ears!

Reply:
You'll end up blocking on the channel, because you don't know exactly when the error will come in, if at all. You should have the error checking, but it should call `something.Cancel()` on error. This may mean you need to put the `Start()` below the addition to `i.extensionCaches[]` below. The `Cancel()` function will remove the item from the caches. This way, `Cancel()` is called regardless.

Reply:
```suggestion
	i.extensionCaches[ce.Name] := extensionCacheData{
		Cache:  c,
		Cancel: cancel,
	}
	 = dataCache
	go func(){
	    err := c.Start(ctx)
	    if err != nil {
	        cancel()
	    }
	}()
```

Reply:
I may need a synchronous chat to better understand how this should be handled. I haven't had a scenario like this before and my brain is failing me on understanding the approach you are suggesting @tmshort